pubtools/exodus/_tasks/push.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import subprocess
+
+from pubtools.exodus.task import ExodusTask
+
+LOG = logging.getLogger("pubtools-exodus")
+LOG_FORMAT = "%(asctime)s [%(levelname)-8s] %(message)s"
+
+
+class ExodusPushTask(ExodusTask):
+    """Push a directory to the Exodus CDN"""
+
+    def __init__(self):
+        super(ExodusPushTask, self).__init__()
+        self.src = None
+        self.dest = None
+
+    def add_args(self):
+        super(ExodusPushTask, self).add_args()
+
+        self.parser.add_argument(
+            "--source",
+            help="The path to the directory to be pushed to the exodus CDN",
+        )
+        self.parser.add_argument(
+            "--dest",
+            help="The destination path to publish to on the exodus CDN",
+        )
+
+    def run(self):
+        LOG.debug("Exodus push begins")
+
+        if not self.publish:
+            self.publish = self.new_publish()
```

Comment:
It seems to me like there's no reason to store several of these objects on 'self'.

self.publish, self.src and self.dest are all used only within this one method, or passed into other methods meaning those methods don't access them via self.  Is there any reason to store them on self?  And why, for example, do we have self.src and not self.cmd.

Usually it's a good idea to keep variables as plain locals if you don't have a reason to make them instance variables, as adding them onto the instance makes them available to hidden side effects occurring elsewhere, making it harder to understand the code.